### PR TITLE
gh-101101: Fix test_code_extra to reset value for refleak test

### PR DESCRIPTION
**Pull request by corona10**
* Issue: gh-101101

### CODE REVIEW

**Comment by corona10** [comment]
With out this:
````

➜  cpython git:(main) ✗ ./python.exe -m test test_capi -R 3:3
Raised RLIMIT_NOFILE: 256 -> 1024
0:00:00 load avg: 3.41 Run tests sequentially
0:00:00 load avg: 3.41 [1/1] test_capi
beginning 6 repetitions
123456
.Assertion failed: (extra == NULL), function test_code_extra, file code.c, line 82.
Fatal Python error: Aborted

Current thread 0x0000000104ba8580 (most recent call first):
  File "/Users/user/oss/cpython/Lib/unittest/case.py", line 579 in _callTestMethod
  File "/Users/user/oss/cpython/Lib/unittest/case.py", line 623 in run
  File "/Users/user/oss/cpython/Lib/unittest/case.py", line 678 in __call__
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 122 in run
  File "/Users/user/oss/cpython/Lib/unittest/suite.py", line 84 in __call__
  File "/Users/user/oss/cpython/Lib/test/support/testresult.py", line 140 in run
  File "/Users/user/oss/cpython/Lib/test/support/__init__.py", line 1104 in _run_suite
  File "/Users/user/oss/cpython/Lib/test/support/__init__.py", line 1230 in run_unittest
  File "/Users/user/oss/cpython/Lib/test/libregrtest/runtest.py", line 281 in _test_module
  File "/Users/user/oss/cpython/Lib/test/libregrtest/refleak.py", line 89 in dash_R
  File "/Users/user/oss/cpython/Lib/test/libregrtest/runtest.py", line 315 in _runtest_inner2
  File "/Users/user/oss/cpython/Lib/test/libregrtest/runtest.py", line 360 in _runtest_inner
  File "/Users/user/oss/cpython/Lib/test/libregrtest/runtest.py", line 235 in _runtest
  File "/Users/user/oss/cpython/Lib/test/libregrtest/runtest.py", line 265 in runtest
  File "/Users/user/oss/cpython/Lib/test/libregrtest/main.py", line 456 in run_tests_sequential
  File "/Users/user/oss/cpython/Lib/test/libregrtest/main.py", line 574 in run_tests
  File "/Users/user/oss/cpython/Lib/test/libregrtest/main.py", line 752 in _main
  File "/Users/user/oss/cpython/Lib/test/libregrtest/main.py", line 711 in main
  File "/Users/user/oss/cpython/Lib/test/libregrtest/main.py", line 775 in main
  File "/Users/user/oss/cpython/Lib/test/__main__.py", line 2 in <module>
  File "/Users/user/oss/cpython/Lib/runpy.py", line 88 in _run_code
  File "/Users/user/oss/cpython/Lib/runpy.py", line 198 in _run_module_as_main

Extension modules: _testcapi, _testmultiphase, _testsinglephase, _testinternalcapi (total: 4)
[1]    83285 abort      ./python.exe -m test test_capi -R 3:3
````

**Comment by corona10** [comment]
> buildbot/s390x RHEL7 Refleaks PR 

Unrelated to this PR.

File: Modules/_testcapi/code.c [hunk 1]
```diff
@@ -92,7 +92,11 @@ test_code_extra(PyObject* self, PyObject *Py_UNUSED(callable))
         goto finally;
     }
     assert ((uintptr_t)extra == 77);
-
+    // Need to reset code extra value.
```
Comment (erlend-aasland) [inline Modules/_testcapi/code.c]:
```suggestion
    // Revert to initial code extra value.
```

**Review by erlend-aasland — COMMENTED** [review]
Small nit to make the comment slightly clearer. Perhaps also add code to assert that we could successfully revert to the initial value?

**Comment by encukou** [comment]
Ah, that's right. The test needs to clean up after itself so it can be run again.
Thank you for the fix!

**Review by encukou — APPROVED** [review]

**Pull request merged**